### ISSUE TITLE
assertionCreator lambda which do not create assertions should fail

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#96]
*Platform* (JVM, JS and/or Android): all

## Code related feature
```
assert(listOf(1, 2)) {
}
```
The above should result in a failure. Currently we throw already IllegalStateException in certain cases where one forgets to define assertions. 

I think we should enforce this in all cases with `Expect` (keep the current behaviour for `Assert`)
However, I would no longer throw an IllegalStateException but create an assertion which fails together with a text explaining that no assertion was created. So that the above fails with something like:
```
assert: List(1, 2)
- at least one assertion defined: false
  >> You forgot to define assertions in the assertionCreator lambda
  >> Sometimes you can use an alternative to `{ }` For instance, instead of `toThrow<..> { }` you should use `toThrow<..>()`
```

As mentioned, this should be used every time one has to pass an assertionCreator lambda. Following another example:
```
assert(listOf(1, 2))
  .get(0) { }
  .get(1) { toBe(1) }
```
Currently we throw and break reporting. I would image that we don't but fail with an error message something like:
```
assert: List(1, 2)
-> get(0): 1 
    - at least one assertion defined: false
      >> You forgot to define assertions in the assertionCreator lambda
      >> Sometimes you can use an alternative to `{ }` For instance, instead of `toThrow<..> { }`  you should use `toThrow<..>()`
- get(1): 2
  - to be: 1
```

We could even try to enforce it during compile time (at least to a certain degree) by defining an assertionCreator lambda to have not `Unit` but `Expect <*>` as return type. Yet, this also places a certain restriction on assertion functions. They all need to have a fluent API. Atrium v0.8.0 does not fulfil that and I guess it is likely that users did not care too much about the return type unless they wanted to use it in a fluent style

**Comment by robstoll** [comment]
Moved to 0.10.0 in 0.9.0 we will make sure it always throws

**Comment by robstoll** [comment]
implemented by 675b3b35
